drivers/console/uart_console.c

```diff
@@ -107,13 +107,36 @@ static int console_out(int c)
 
 	return c;
 }
+#endif
+
+#if defined(CONFIG_STDIN_CONSOLE)
+/**
+ *
+ * @brief Get one character from UART
+ *
+ * @return The character received from console
+ */
+static int console_in(void)
+{
+	unsigned char c;
 
+	while (uart_poll_in(uart_console_dev, &c) < 0) {
+		k_msleep(2);
```

Comment:
See [`CONFIG_SHELL_BACKEND_SERIAL_RX_POLL_PERIOD`](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_SHELL_BACKEND_SERIAL_RX_POLL_PERIOD)

Reply:
Hi. Should I use this Kconfig option somehow, or create a similar Kconfig option, or just use its default value that is 10ms in this code.
Can you guide?

Reply:
Added Kconfig option UART_RX_POLL_PERIOD in drivers/console/Kconfig

Thanks

Reply:
Hi @ycsin ! Could you take a look at my PR? I'm open to any feedback and happy to make any changes you suggest. Thanks!